internal/rbd/rbd_util_test.go

```diff
@@ -42,3 +44,86 @@ func TestHasSnapshotFeature(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateImageFeatures(t *testing.T) {
+	tests := []struct {
+		imageFeatures string
+		rbdVol        *rbdVolume
+		isErr         bool
+		errMsg        string
+	}{
+		{
+			"layering",
+			&rbdVolume{
+				Mounter: rbdDefaultMounter,
+			},
+			false,
+			"",
+		},
+		{
+			"layering",
+			&rbdVolume{
+				Mounter: rbdNbdMounter,
+			},
+			false,
+			"",
+		},
+		{
+			"layering,exclusive-lock,journaling",
+			&rbdVolume{
+				Mounter: rbdNbdMounter,
+			},
+			false,
+			"",
+		},
+		{
+			"layering,journaling",
+			&rbdVolume{
+				Mounter: rbdNbdMounter,
+			},
+			true,
+			"feature journaling requires exclusive-lock",
+		},
+		{
+			"layering,exclusive-lock,journaling",
+			&rbdVolume{
+				Mounter: rbdDefaultMounter,
+			},
+			true,
+			"feature exclusive-lock requires rbd-nbd for mounter",
+		},
+		{
+			"layering,exclusive-lock,journaling",
+			&rbdVolume{
+				Mounter: rbdDefaultMounter,
+			},
+			true,
+			"feature exclusive-lock requires rbd-nbd for mounter",
+		},
+		{
+			"layering,exclusive-loc,journaling",
+			&rbdVolume{
+				Mounter: rbdNbdMounter,
+			},
+			true,
+			"invalid feature exclusive-loc",
+		},
+		{
+			"ayering",
```

Comment:
```suggestion
			"layering",
```

Reply:
This is intentional, an error is expected